### PR TITLE
Backport of https://github.com/containers/podman/pull/13536 to 3.4

### PR DESCRIPTION
**Pull request by dcermak**

### CODE REVIEW

**Review by vrothberg — COMMENTED** [review]
Something must have gone wrong, a `+15,343 −4,498` backport seems rather risky. Looking at all the bumps in dependencies (storage even to a non-release), I think some other backports are required before - assuming this backport is needed.

**Comment by dcermak** [comment]
> Something must have gone wrong, a `+15,343 −4,498` backport seems rather risky. Looking at all the bumps in dependencies (storage even to a non-release), I think some other backports are required before - assuming this backport is needed.

I believe that a backport is needed, as I mentioned in https://github.com/containers/podman/pull/13536#issuecomment-1073865883.

I can cherry pick the actual fix into the https://github.com/containers/storage/tree/release-1.37 branch and then vendor that one?

**Comment by vrothberg** [comment]
> I can cherry pick the actual fix into the https://github.com/containers/storage/tree/release-1.37 branch and then vendor that one?

Yes, that would be great, thank you!

**Comment by mheon** [comment]
LGTM once the storage change merges

**Comment by dcermak** [comment]
Could someone restart the CI? The tests failed with an unrelated infra issue:
```
# podman pull quay.io/libpod/testimage:20210610
Trying to pull quay.io/libpod/testimage:20210610...
Error: initializing source docker://quay.io/libpod/testimage:20210610: Requesting bear token: invalid status code from registry 500 (Internal Server Error)
```

File: go.mod [hunk 1]
```diff
@@ -17,7 +17,7 @@ require (
 	github.com/containers/image/v5 v5.17.0
 	github.com/containers/ocicrypt v1.1.2
 	github.com/containers/psgo v1.7.1
-	github.com/containers/storage v1.37.0
+	github.com/containers/storage v1.37.1-0.20220323114520-095fcc2bfaa8
```
Comment (vrothberg) [inline go.mod:20]: I'd prefer to cut a v1.37 and vendor that over vendoring a commit.

Comment (rhatdan) [reply go.mod:20]: I agree.

Comment (dcermak) [reply go.mod]: Done

**Comment by dcermak** [comment]
@rhatdan The `Optional Release Test` CI test fails with:
```
ERROR: Releases must never contain '-dev' in output of 'podman info' (  APIVersion: 3.4.5-dev
  Version: 3.4.5-dev)  (contrib/cirrus/runner.sh:268 in _run_release())
```
Should I look into fixing this?

**Comment by rhatdan** [comment]
Can you run podman -v and see if this shows the -dev flag, if not, then I would not worry about it.

**Comment by dcermak** [comment]
> Can you run podman -v and see if this shows the -dev flag, if not, then I would not worry about it.

It shows it.

**Comment by Luap99** [comment]
The "Optional Release Test" is an optional test as the name states, it must be triggered manually. However running it on random PRs is wrong, it is only designed to be run on release PRs to make sure the version is correctly set.

The result can be ignored.

/hold cancel

**Pull request merged**